include/quic/btstream.hpp

```diff
@@ -143,8 +187,27 @@ namespace oxen::quic
 
         void process_incoming(std::string_view req);
 
-        std::shared_ptr<sent_request> make_command(
-                std::string endpoint, std::string body, std::function<void(message)> = nullptr);
+        template <typename... Opt>
+        std::shared_ptr<sent_request> make_command(std::string endpoint, std::string body, Opt&&... opts)
+        {
+            oxenc::bt_list_producer btlp;
+            auto rid = ++next_rid;
+
+            try
+            {
+                btlp.append("C");
+                btlp.append(rid);
+                btlp.append(endpoint);
+                btlp.append(body);
+
+                return std::make_shared<sent_request>(*this, std::move(btlp).str(), rid, std::forward<Opt>(opts)...);
+            }
+            catch (const std::exception& e)
+            {
+                log::critical(bp_cat, "Invalid outgoing command encoding: {}", e.what());
+                return nullptr;
```

Comment:
```suggestion
                throw;
```
so that we log, but we still propagate the original exception.